parsl/executors/taskvine/executor.py

```diff
@@ -493,6 +589,12 @@ def _serialize_object_to_file(self, path, obj):
             while written < len(serialized_obj):
                 written += f_out.write(serialized_obj[written:])
 
+    def _cloudpickle_serialize_object_to_file(self, path, obj):
```

Comment:
we talked about this somewhere before but I can't remember where: you should be using the parsl serialization libraries not cloudpickle unless you have a specific reason that needs different serialization.

Reply:
yeah the `__main__` module is one (maybe the only?) place where pickle doesn't behave as a strict subset of dill.

I'm kinda inclined to change the parsl.serialize behaviour to not do pickle first - because behaviour like you report above with `__main__` is usually wrong anyway, as far as a Parsl-style distributed environment.